### DISCUSSION TITLE
Cleaning before OCR

### DISCUSSION THREAD

**Discussion by fufurax** [ImageMagick/ImageMagick#5660] [Help]
Hi,
For personnal project i try to get a number on screen shot from ecycling application.
I've write swall application in C# who takes screen shot and use Tesseract to read text but I've poor result quality.

my process 

1. take screen shot (1366 * 768) : https://ibb.co/Hx5PZDL
2. crop (57*72) : https://ibb.co/K2YfVGT
3. "cleaning colors" (57*72) : https://ibb.co/dfj5MTD
```
 ImageScreen.ColorFuzz = new Percentage(10);
                    ImageScreen.Transparent(_Zcolors[0]);
                    ImageScreen.Transparent(_Zcolors[1]);
                    ImageScreen.Transparent(_Zcolors[2]);
                    ImageScreen.Transparent(_Zcolors[3]);
                    ImageScreen.Alpha(AlphaOption.Extract);
                    ImageScreen.Negate();

                    ImageScreen.Transparent(MagickColors.Black);
                    ImageScreen.Alpha(AlphaOption.Extract);
                    ImageScreen.Negate();
                    ImageScreen.Threshold(new Percentage(15));
```
4. compose to 300*300 >> https://ibb.co/mSwzD6L
            ```
MagickImage Overlay = new MagickImage("xc:white", 300, 300);
            Overlay.Density = Main.Density;
            Overlay.Transparent(MagickColors.White);
            Overlay.Composite(Main, CompositeOperator.SrcOver);

```
5.  and send it to tesseract. 

I'm not specilist on the subject i think there is probably solution in imageMagick to clean and optimize the process.

Thanks for your help

**Comment by fmw42** [comment]
See my textcleaner script at https://github.com/dlemstra/ImageMagick.FredsScripts.NET

**Reply by fufurax** [reply]
I've try :) Look to my issue :) 

https://github.com/dlemstra/ImageMagick.FredsScripts.NET/issues/29
